docs/building/forms.md

```diff
@@ -0,0 +1,107 @@
+A form is like a collection of menus or raw responses and it is basically collecting information from the user.
+
+It is composed by a **header**, a **body** and a **footer**. The body represents form items and these can be slightly modified menus or raw responses.
+
+The form items are being processed in a successive manner. After all the form items have been processed, the user can confirm his choices or reset any of them.
+
+## Json Structure
+
+Your server must return a JSON response with a structure similar to the one below:
+
+```
+{'body': [{'type': 'string',
+           'name': 'descr',
+           'description': 'Provide a description',
+           'method': 'POST',
+           'path': '/callback-path'},
+          {'type': 'date'
```

Comment:
On this and below item `method` key is missing. Was that on purpose?

Reply:
Yes. Thats optional, its explained in the body section

Reply:
> **method** - indicates how the HTTP request will be made (GET, POST, etc...)

The "optional" part is not specified. However, looking at the examples, on a first view it seems there is no structure. Maybe we should make the values null for optional fields, but keep the keys to make the structure clear. What do you think?

Reply:
There is a structure, but not that crystal clear. Thats an example and it probably should have come after the keys explained section. Will tidy up

Reply:
I added here the openapi schema:https://app.swaggerhub.com/apis/romeo1m/schemajson/m1 (m is for Marius :smile: ). Maybe it should be inserted in docs somehow. @chrishornmem what do you think?

Reply:
Yes, thats perfect in my opinion. I will add it I the docs